sdk/src/Core/Amazon.Runtime/Internal/Auth/AWS4Signer.cs

```diff
@@ -886,6 +887,30 @@ protected internal static string CanonicalizeHeaders(IEnumerable<KeyValuePair<st
             return builder.ToString();
         }
 
+        /// <summary>
+        /// Computes the canonical headers with values for the request. Only headers included in the signature
+        /// are included in the canonicalization process.
+        /// </summary>
+        /// <param name="sortedHeaders">All request headers, sorted into canonical order</param>
+        /// <param name="builder">The value string builder to append to.</param>
+        /// <returns>Canonicalized string of headers, with the header names in lower case.</returns>
+        protected internal static void AppendCanonicalizedHeaders(ICollection<KeyValuePair<string, string>> sortedHeaders, ref ValueStringBuilder builder)
```

Comment:
I'll need to do some research on why this was made protected but we don't want to expose `ValueStringBuilder` as part of the external API.

Reply:
Since I haven't received more feedback I'm going with a lighter variant that will use ValueStringBuilder internally within the methods only. The impact should already be quite high and this will leave the API surface untouched for now.